### PR TITLE
Implement numpy.ndarray.itemset

### PR DESCRIPTION
**Pull request by oishigyunyu**
Hello.

I've implemented `numpy.ndarray.itemset` in CuPy referenced at issue of #7414 .

I wrote the test code and tested for 3 diffrent shepe arrays both of cupy and numpy.

Could you please check it?

Thank you.

### CODE REVIEW

File: cupy/_core/core.pyx [hunk 1]
```diff
@@ -485,7 +485,32 @@ cdef class _ndarray_base:
         """
         return self.get().tolist()
 
-    # TODO(okuta): Implement itemset
+    
+    cpdef itemset(self, index, value):
```
Comment (kmaehashi) [inline cupy/_core/core.pyx:489]: `itemset` should be callable with one argument (if size of self is 1). Use `*args` here.

Basically, this method must behave equivalently to the following NumPy code:
https://github.com/numpy/numpy/blob/ea0b1708b27e683201285da53c6afda24bc8ba74/numpy/core/src/multiarray/methods.c#L2985-L2987
https://github.com/numpy/numpy/blob/ea0b1708b27e683201285da53c6afda24bc8ba74/numpy/core/src/multiarray/methods.c#L768

File: tests/cupy_tests/indexing_tests/test_itemset.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    {'shape': (3, 256, 256), 'index': (0, 100, 200), 'value': 5},
+    {'shape': (20, 3, 128, 128), 'index': (560), 'value': 255},
+    {'shape': (20, 3, 128, 128), 'index': (2, 0, 19, 34), 'value': 5},
+)
+@testing.gpu
+class TestItemset(unittest.TestCase):
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_itemset(self, xp, dtype):
```
Comment (kmaehashi) [inline tests/cupy_tests/indexing_tests/test_itemset.py:23]: Please cover the following cases:
* shape = 0-dim (size = 1)
* shape = 1-dim (size = 1)
* dtype of array and value are different
* exception cases

**Review by kmaehashi — CHANGES_REQUESTED** [review]

File: cupy/_core/core.pyx (hunk 1 above)
Comment (oishigyunyu) [reply cupy/_core/core.pyx:489]: Thank you for checking my code.
I've forgotten the case of if size of self is 1.
I will fix this and also test code.

**Comment by kmaehashi** [comment]
As discussed in #7569 this PR is open to everyone to takeover.

**Pull request closed without merging**